src/vocutouts/workers.py

```diff
@@ -75,111 +86,87 @@ class TaskTransientError(Exception):
-    # of the output collection.
-    uuid = uuid4()
-    positions_collection = f"service/cutouts/positions/{uuid}"
-    output_collection = f"service/cutouts/{uuid}"
-    repository = os.environ["CUTOUT_BUTLER_REPOSITORY"]
-    collection = os.environ["CUTOUT_BUTLER_COLLECTION"]
-    butler = Butler(repository, run=positions_collection)
-
-    # This should only need to be done once, but it does have to be done once.
-    # For now, register the dataset type each time, but this is silly so we
-    # should come up with a better initialization process.  (Maybe do this as
-    # part of vo-cutouts init.)
-    dataset_type = DatasetType(
-        "cutout_positions",
-        dimensions=data_id.keys(),
-        universe=butler.registry.dimensions,
-        storageClass="AstropyQTable",
-    )
-    butler.registry.registerDatasetType(dataset_type)
-
-    # Store the cutout parameters in the input Butler collection.
-    #
-    # Use a conversion ratio of 0.2 arcsec/pixel so that we can use the
-    # current test cutout pipeline.  The eventual cutout pipeline will accept
-    # min/max pairs in ICRS coordinates.
-    if any(math.isinf(v) for v in (ra_min, ra_max, dec_min, dec_max)):
-        raise TaskFatalError("UsageError Unbounded ranges not yet supported")
-    pos = SkyCoord(ra_min, dec_min, unit="deg")
-    xspan = ((ra_max - ra_min) * 3600 / 0.2) * u.dimensionless_unscaled
-    yspan = ((dec_max - dec_min) * 3600 / 0.2) * u.dimensionless_unscaled
-    row = 1 * u.dimensionless_unscaled
-    input_table = QTable(
-        [[row], [pos], [xspan], [yspan]],
-        names=["id", "position", "xspan", "yspan"],
-    )
-    butler.put(input_table, "cutout_positions", **data_id)
-
-    # Perform the cutout.  This currently uses subprocess to call pipetask.
-    # The long-term plan is for there to be an API for running a pipeline so
-    # that any setup overhead can be incurred once at worker start and then
-    # cutouts can be done with a direct Python call.
-    #
-    # This tells Butler to register data types every time, which as above
-    # should not be necessary, but they do have to be registered once.
-    data_query_terms = []
-    for key, value in data_id.items():
-        if isinstance(value, int):
-            data_query_terms.append(f"{key}={value}")
+    # Currently, only a single data ID and a single stencil are supported.
+    assert len(dataset_ids) == 1
+    assert len(stencils) == 1
+
+    # Convert the stencils to SkyStencils.
+    sky_stencils: List[SkyStencil] = []
+    for stencil_dict in stencils:
+        if stencil_dict["type"] == "circle":
+            center = SkyCoord(
+                stencil_dict["center"]["ra"] * u.degree,
+                stencil_dict["center"]["dec"] * u.degree,
+                frame="icrs",
+            )
+            radius = Angle(stencil_dict["radius"] * u.degree)
+            stencil = SkyCircle.from_astropy(center, radius)
```

Comment:
Consider a SkyCircle.from_dict (or add a protocol to the SkyStencils base class?) so that you don't need to mess with the JSON-able dict to Astropy conversion work here.

Reply:
Oh, perhaps SkyCircle is in the wrong layer of responsibility for that to be reasonable.

Reply:
Yeah, SkyCircle belongs to the backend. I thought about asking Jim to add support for a dict representation, but it's kind of a weird internal representation that exists only because Dramatiq needs something that's JSON-serializable, and I'm not sure it makes sense on the backend. We're asserting here that we know these are IRCS coordinates in degrees, etc., which are things that I think the backend reasonably doesn't want to assume. I agree that the conversion code is a bit awkward, but I couldn't see a clearly better way to do it.